src/arch/x86/lib/cpu_performance/cpu_info.c

```diff
@@ -21,10 +23,18 @@ static void set_cpu_features(struct cpu_info *info) {
 					: "=g"(reg)
 					:
 					: "%eax", "%ebx", "%ecx", "%edx");
-
-	info->FPU = reg & 0x1;
-	info->MMX = (reg >> 23) & 0x1; 
-	info->SSE = (reg >> 25) & 0x1;
+	
+	assert(current_feature < CURRENT_ARCH_NUM_FEATURES);
+	strcpy(info->feature[current_feature].name, "FPU");
+	info->feature[current_feature++].val = reg & 0x1;
+	
+	assert(current_feature < CURRENT_ARCH_NUM_FEATURES);
+	strcpy(info->feature[current_feature].name, "MMU");
+	info->feature[current_feature++].val = (reg >> 23) & 0x1;
+	
+	assert(current_feature < CURRENT_ARCH_NUM_FEATURES);
+	strcpy(info->feature[current_feature].name, "SSE");
```

Comment:
Why not create function to avoid copying of `assert/strcpy/current_feature++`? I mean it may look like this

```
int set_cpu_feature(struct cpu_info *info, const char *name, unsigned val) {
  int current_feature = info->feature_count;
  /* todo: sanity checks for argument */
  strcpy(info->feature[current_feature], name);
  info->feature[current_feature] = val;
  info->feature_count++;
}

set_cpu_feature(info, "FPU", reg & 0x1);
set_cpu_feature(info, "MMU", (reg >> 23) & 0x1);
set_cpu_feature(info, "SSE", (reg >> 25) & 0x1);
```

Reply:
Have tried to implement something of the sort in the latest fixup. Please review, thanks :)